### PR TITLE
Add attribution option to plotting functions

### LINKED ISSUE #71: verbose geom_phylopic
Add an argument (`verbose`, maybe?) to `geom_phylopic` that spits out the attribution information for the silhouettes (mostly useful when `name = ` is used. Perhaps there should also be a warning when `name = ` is used and `verbose = FALSE`?

### PR DESCRIPTION
**Pull request by willgearty**
This adds the `verbose` argument to our three plotting functions (geom_phylopic, add_phylopic, and add_phylopic_base) that outputs the get_attribution text for the silhouettes used. If the `name` aesthetic/argument is used in these functions (implying the user doesn't have a list of uuids or image objects in their environment), a warning is printed if `verbose` is not set to `TRUE`.

Fixes #71.

### CODE REVIEW

**Comment by keesey** [comment]
Any thoughts about utilizing the `/collections` endpoints on _PhyloPic_ for attribution? You can link to pages like these: https://www.phylopic.org/collections/39f27cb5-5952-01d2-002a-0721a815ac14

There is also a way to turn those into permalinks, although I still need to document that for public usage.

**Comment by willgearty** [comment]
@keesey Ah, I had always thought you needed to create collections in the web portal, but I see now you can generate them with the API. We'll look into incorporating that!

**Comment by willgearty** [comment]
@keesey Are collection uuids essentially permalinks? In other words, will a collection URL always include the same list of images?

File: R/add_phylopic_base.r [hunk 1]
```diff
@@ -101,17 +103,22 @@ add_phylopic_base <- function(img = NULL, name = NULL, uuid = NULL,
     if (!is.character(name)) {
       stop("`name` should be of class character.")
     }
+    if (!verbose) {
+      warning(paste("You've used the `name` argument. You may want to use",
```
Comment (LewisAJones) [inline R/add_phylopic_base.r:107]: Oppps nevermind, this covers my above comment well! Good job.

File: R/add_phylopic_base.r [hunk 2]
```diff
@@ -86,7 +88,7 @@ add_phylopic_base <- function(img = NULL, name = NULL, uuid = NULL,
                               x = NULL, y = NULL, ysize = NULL,
                               alpha = 1, color = "black",
                               horizontal = FALSE, vertical = FALSE, angle = 0,
-                              remove_background = TRUE) {
+                              remove_background = TRUE, verbose = FALSE) {
```
Comment (LewisAJones) [inline R/add_phylopic_base.r:91]: Error handling for verbose?

File: R/add_phylopic_base.r [hunk 3]
```diff
@@ -101,17 +103,22 @@ add_phylopic_base <- function(img = NULL, name = NULL, uuid = NULL,
     if (!is.character(name)) {
       stop("`name` should be of class character.")
     }
+    if (!verbose) {
+      warning(paste("You've used the `name` argument. You may want to use",
+                    "`verbose = TRUE` to get attribution information",
+                    "for the silhouettes."), call. = FALSE)
```
Comment (LewisAJones) [inline R/add_phylopic_base.r:109]:
```suggestion
                    "for the silhouette(s)."), call. = FALSE)
```

File: R/geom_phylopic.R [hunk 4]
```diff
@@ -113,20 +120,25 @@ GeomPhylopic <- ggproto("GeomPhylopic", Geom,
 
     # Check supplied data types and retrieve silhouettes if need be
     if (cols["name"]) {
+      if (!verbose) {
+        warning(paste("You've used the `name` aesthetic/argument. You may want",
+                      "to use `verbose = TRUE` to get attribution information",
+                      "for the silhouettes."), call. = FALSE)
```
Comment (LewisAJones) [inline R/geom_phylopic.R:126]:
```suggestion
                      "for the silhouette(s)."), call. = FALSE)
```

File: R/add_phylopic.r [hunk 5]
```diff
@@ -25,6 +25,8 @@
 #'   clockwise. The default is no rotation.
 #' @param remove_background \code{logical}. Should any white background be
 #'   removed from the silhouette(s)? See [recolor_phylopic()] for details.
+#' @param verbose \code{logical}. Should the attribution information for the
+#'   used silhouettes be printed to the console (see [get_attribution()])?
```
Comment (LewisAJones) [inline R/add_phylopic.r:29]:
```suggestion
#'   used silhouette(s) be printed to the console (see [get_attribution()])?
```

**Review by LewisAJones — CHANGES_REQUESTED** [review]
Hi @willgearty, this is looking pretty good and almost there. The main issue I see is that with `geom_phylopic` the warning message and attribution information is called for each unique name, see example: 

```
library(ggplot2)
df <- data.frame(x = c(2, 4), y = c(10, 20), name = c("cat", "walrus"))
# Multiple warning messages
ggplot(df) +
  geom_phylopic(aes(x = x, y = y, name = name),
                verbose = FALSE, color = "purple", size = 10) +
  facet_wrap(~name) +
  coord_cartesian(xlim = c(1,6), ylim = c(5, 30))
# Multiple attribution messages (should just be one).
ggplot(df) +
  geom_phylopic(aes(x = x, y = y, name = name),
                verbose = TRUE, color = "purple", size = 10) +
  facet_wrap(~name) +
  coord_cartesian(xlim = c(1,6), ylim = c(5, 30))
```

**Comment by willgearty** [comment]
I think it's once per facet? I figured this was fine because often you list separate captions for each facet/panel of a figure anyways?

**Comment by LewisAJones** [comment]
> I think it's once per facet/panel for geom_phylopic? I figured this was fine because often you list separate captions for each facet/panel of a figure anyways?

Maybe? I can just imagine a situation where you have many facets and you flood the console (like our LBG paper for example).

**Comment by willgearty** [comment]
Hmm...that might require a complete refactoring of GeomPhylopic, then. We currently process the names/uuids/imgs on a per-panel basis (this is pretty standard for geoms), so we would need to move all of this processing to a different step of the geom construction process...

**Comment by LewisAJones** [comment]
> Hmm...that might require a complete refactoring of GeomPhylopic, then. We currently process the names/uuids/imgs on a per-panel basis (this is pretty standard for geoms), so we would need to move all of this processing to a different step of the geom construction process. I'll look into it...

Sounds like an update for a later version. Happy with moving forward as is!

**Comment by willgearty** [comment]
I decided to go ahead and make the change anyways. Let me know what you think @LewisAJones!

**Comment by keesey** [comment]
> @keesey Are collection uuids essentially permalinks? In other words, will a collections/ URL always include the same list of images?

It will always include the same list of UUIDs, but it's always possible that images will be removed or details changed. (Although both of those are quite infrequent.) Collections can be turned into actual permalinks, which will never change, but I haven't moved that to the general API yet. (Not sure when I'll have bandwidth.)

File: R/add_phylopic.r [hunk 6]
```diff
@@ -72,7 +74,7 @@ add_phylopic <- function(img = NULL, name = NULL, uuid = NULL,
                          x, y, ysize = Inf,
                          alpha = 1, color = "black",
                          horizontal = FALSE, vertical = FALSE, angle = 0,
-                         remove_background = TRUE) {
+                         remove_background = TRUE, verbose = FALSE) {
```
[the remark this answers is not in the document]
Comment (willgearty) [reply R/add_phylopic.r:86]: I totally thought about this as well. I think a message on package load would be great! (but let's make that a separate issue)

**Review by LewisAJones — APPROVED** [review]
All seems good to me now @willgearty, thanks for addressing my previous concern in this PR! I hope it wasn't too much extra work.

**Pull request merged**